### PR TITLE
Bump calico and canal to v3.26.1

### PR DESCRIPTION
**Pull request by doflamingo721**
Linked issue: https://github.com/rancher/rancher/issues/41840

### CODE REVIEW

File: pkg/rke/templates/calico_v3.26.go [hunk 1]
```diff
@@ -0,0 +1,5073 @@
...
+            - mountPath: /var/lib/cni/networks
+              name: host-local-net-dir
+            - mountPath: /host/opt/cni/bin
+              name: cni-bin-dir
+          securityContext:
+            privileged: true
+        # This container installs the CNI binaries
+        # and CNI network config file on each node.
+        - name: install-cni
+          image: docker.io/calico/cni:v3.26.1
```
Comment (jiaqiluo) [inline pkg/rke/templates/calico_v3.26.go]: need to be done at multiple places 
```suggestion
          image: {{.CNIImage}}
```

File: pkg/rke/templates/canal_v3.26.go [hunk 2]
```diff
@@ -0,0 +1,5056 @@
+package templates
+
+/*
+CanalTemplateV3_26_1 is based on upstream canal v3.26.1
+https://raw.githubusercontent.com/projectcalico/calico/v3.25.0/manifests/canal.yaml
```
Comment (jiaqiluo) [inline pkg/rke/templates/canal_v3.26.go]:
```suggestion
https://raw.githubusercontent.com/projectcalico/calico/v3.26.1/manifests/canal.yaml
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: pkg/rke/templates/calico_v3.26.go (hunk 1 above)
Comment (rayandas) [reply pkg/rke/templates/calico_v3.26.go]: Fixed it

File: pkg/rke/templates/canal_v3.26.go (hunk 2 above)
Comment (rayandas) [reply pkg/rke/templates/canal_v3.26.go]: Updated the PR

**Review by jiaqiluo — DISMISSED** [review]
LGTM

**Comment by rayandas** [comment]
@jiaqiluo Sorry, earlier template wasn't working. So I did a fresh template update and tested it and it's working now. Could you please review once?
cc: @kinarashah @doflamingo721

**Review by jiaqiluo — APPROVED** [review]

**Review by kinarashah — APPROVED** [review]

**Pull request merged**